src/python/scriptless_zkp/ecc/zkp/nizk_equal_dlogs_proof.py

```diff
@@ -251,23 +254,37 @@ def from_string_encoding(cls, encoded_parameter_set: str) -> NIZKDiscreteLogPara
                 dlog_base_bytes: bytes = base64.b64decode(dlog_base_field)
                 ref_pt_bytes: bytes = base64.b64decode(ref_pt_field)
 
-                dlog_base = ECC.import_key(dlog_base_bytes, curve_name=curve_name)
-                ref_pt = ECC.import_key(ref_pt_bytes, curve_name=curve_name)
+                # Decode the ECC discrete log's assoc. base point from its SEC1 binary encoding.
+                dlog_base: ECC.EccPoint = ECC.import_key(
+                    dlog_base_bytes,
+                    curve_name=curve_name
+                ).pointQ
+                # Decode the ECC discrete log's assoc. "reference" point from its SEC1 binary encoding (i.e., where this
+                # "reference" point equals the base point multiplied by the discrete log).
+                ref_pt: ECC.EccPoint = ECC.import_key(
+                    ref_pt_bytes,
+                    curve_name=curve_name
+                ).pointQ
```

Comment:
Good catch, on the missing conversion to an EccPoint for the d-log reference point.